esp/platform/persistent.cpp

```diff
@@ -47,33 +47,35 @@ class CPersistentInfo : implements IInterface, public CInterface
     unsigned useCount;
     std::unique_ptr<SocketEndpoint> ep;
     StringBuffer epstr;
+    StringBuffer keystr;
     PersistentProtocol proto;
-    ISocket* sock;
+    Linked<ISocket> sock;
 };
 
-using SocketSet = std::unordered_set<CPersistentInfo*>;
-using EpSocketSetMap = MapStringTo<SocketSet*, SocketSet*>;
+struct LinkedPersistentInfoHash
+{
+    size_t operator()(const Linked<CPersistentInfo>& linkedinfo) const
+    {
+        return std::hash<CPersistentInfo*>()(linkedinfo);
+    }
+};
 
-class CAvailKeeper : implements IInterface, public CInterface
+using SocketSet = std::unordered_set<Linked<CPersistentInfo>, LinkedPersistentInfoHash>;
```

Comment:
Changing it to Owned is causing a ref count issue. I think it may have something to do with how unordered_set copies or returns elements, and how Owned deals with it. I'll investigate more and see if I can find the issue.

Reply:
Did you LINK the CPersistentInfo pointer when inserting?

Reply:
Yes I did.

Reply:
Found the issue. Turned out, socketset->erase(sockinfo) has to be socketset->erase(LINK(sockinfo)), which makes sense as the pointer will be turned into Owned.

Reply:
hm ok, yes makes some sense, but it's ugly/not nice to need link for reasons like this.
In that case, it would be cleaner to have the elements as Linked<> items.
Performance difference is going to be v. small

Reply:
Agree, Linked does fit better in this case.